### ISSUE TITLE
Handling PD amendments / changes in PMP

### ISSUE THREAD

**Issue by chachra** [unicef/etools-partner-reporting-portal#365]
PD could change in PMP. Basic details could change or more advanced things like more outputs, indicators being added or disaggregations being edited etc. Its tough to know what all might change.

**Solution idea**
If a PD change is detected (might require PMP to expose a "version" maybe?) then create a new PD in keep past reports associate with this PD.

**Technical details**
Don't sync result structure till PD == active.

draft -> signed -> **active** -> ended -> closed.

signed->suspended->signed.
active->suspended->active is possible... suspended is a transitory state.

Amendment can be made when its signed or active, and it remains in that status.

One option is to re-associate past progress reports to this new PD. Since old reportables etc. with their own disaggregations will be there, viewing/editing past reports will continue as per normal.

Another option is to create a reference from old PD to new one (parent/child) and hence by that relation all PR's of old PD will belong to new PD. Just like above option viewing/editing will continue per usual. Old PD will be marked inactive and there just for reference purposes. Will maintain history of which report was filed/created against which PD version etc. which might be handy.

Maybe on PRP side keep track of a hash of the structure itself and create a new PD / structure when it changes. To optimize we could strip out things that don't matter structurally.

In PMP LLO could be added/deleted, indicator on a LLO cannot be edited, just removed and new ones added. So in that sense if an new LLO/indicator shows up we sync it and new LLO / disaggs / reportable created and new PR's will have it in the future. And if a indicator gets dropped on a LLO then we mark it as inactive and in the future PR's no IR's for that indicator would be created.

**Comment by chachra** [comment]
CC @ewheeler  @robertavram @dfryta thoughts welcome. Just something we'll have to think of soon as an edge case.

**Comment by chachra** [comment]
Please lmk thoughts on how to handle this...

**Comment by dfryta** [comment]
I would go with second approach (parent_id in PD). That will build up a history of changes and keep data required for creating reports (in case of analysis).

**Comment by dfryta** [comment]
Blocked by lack of access to PMP API.

Database changes:
1. Added `active` to LowerLevelOutput

PD workflow:
1. PD is created regardless of `status`
2. `expected_results` are added only if PD is in `active` status

LLO workflow:
1. If LLO does not exists yet:
-- create LLO
-- create bluperint
-- create locations
-- create/assign disaggregations
-- create/assign disaggregations values
-- create reportable
2. If LLO does not exists anymore:
--  set LLO `active=False`

**Comment by dfryta** [comment]
PMP API works again. @chachra how we should map PMP PD statuses to PRP ones:
- from PMP: draft, signed,  active, suspended, ended, closed
- from PRP:  draft, active, implemented, rejected

**Comment by chachra** [comment]
No reason to have different statuses @dfryta ... please re-align them. They might have changed over time.

**Comment by chachra** [comment]
Meaning in PRP use same as PMP. Make sure to test the UI and nothing else breaks. In our world I'd rather not show ones that are draft, rest can show since those might have reports hanging off them.

**Comment by dfryta** [comment]
Sounds good. I will update PD statuses, fix filtering etc.
